src/battle-animations.ts

```diff
@@ -103,6 +103,8 @@ class BattleScene {
 			else if (pokemonId.charAt(2) === ':') return pokemonId.slice(3).trim();
 			return '???pokemon:' + pokemonId + '???';
 		};
+		
+		/* Digimon Showdown Animation Code */
```

Comment:
So yeah, as @Zarel said, mod code should be *separate* from the main code, not simply surrounded by comments. 

You want to take all the Digimon related code and put it under `src/digimon`. You really have two options:

1) you literally copy all the main files that youve placed digimon code in to the new sub directory and revert the changes in the main files such that you have two mostly parallel source trees and then in the build logic choose to use either the regular code or the digimon code. I feel this is kind of what Zarel has been saying, and would involve the '5 line of glue code' in the build file to make it happen, and is least intrusive to the regular path. This is also the least amount of work in the short term, but quickly becomes undesirable as the large amounts of duplicated code will diverge or require high maintenance costs.
2) mimic the simulator's `data/mods` architecture: move the  digimon specific all into its own file, and add a function call 'hook' in the main client to any mod code that may exist. This is slightly more intrusive because you need a method call at each of the places your digimon code currently resides as well as the glue code to specify which mod is being used (if any), but actually scales and results in the least total code overall.

TBH, I wonder if it would be simpler just to fork and serve the client yourself. Presumably the login server prevents this? **I wonder if finding a way to make it easier to people to host their own forked clients would make more sense than checking this into main.**

Reply:
I’ve suggested the former but they seemed to refuse. The latter would be a great though imo

Reply:
I'd definitely prefer option 2, but I said I'd also accept option 1.

Something like OAuth might also work.

Reply:
Hi, ive been speaking to Alfastorm who is our current head client coder. He wants to do option 1, I trust him so that's what we are going for. He is finishing up school stuff at the moment so this will take some time. I'll keep you posted if we have any hicups or anything.